guides/v2.2/extension-dev-guide/versioning/dependencies.md

```diff
@@ -84,8 +85,9 @@ Use this table to set the appropriate version dependency on a module based on ho
 | | Extend from a class | MINOR|
 | | Override a method | MINOR|
 | | Subscribe to an event| MINOR|
-| **Virtual Type**| Configure a virtual type in the `di.xml` file as a class dependency | MAJOR|
-| **URL Paths** | Link to from custom pages | MAJOR|
+| **Virtual Type (marked with `@api`)**| Use an existing virtual type in the `di.xml` file as a class dependency | MAJOR|
+| **URL Paths** | Link to a URL from custom pages | MAJOR|
+| | Submit a POST request to the URL | MAJOR|
```

Comment:
I still don't agree that changing POST params is a MAJOR change. In most cases, such requests are handled by controllers which are not public API.

Reply:
Controller is private code in terms that nobody should call `Controller->execute()` directly. But an HTTP request is public API (or at least it should be considered as a different thing).

Reply:
What level of change do you suggest for POST params? Then probably, for URLs as well (because somebody can't just rely on a POST URL, while not specifying the params).
MINOR or PATCH?

Reply:
We have clear separation on URL level for WEB API and other requests. Changing POST params WEB API requests should be MAJOR (such entry-points exposed by service contracts), other POST requests - PATCH.

Reply:
Defining any change in any POST request structure as MAJOR it's a very strong restriction.
How about new optional POST parameters? I still believe it should be a PATCH.
Extension developers should verify correct work of their extensions with the lowest patch version of the compatible minor release. In such way, they'll avoid problems with relying on some code that can appear in a patch release (approved or hidden BIC).

Reply:
Another point regarding why all POST HTTP requests should be considered as public API?
I think the statement that somebody can create dependency on them is not enough.
In the earlier stages, I see that it is more of a necessity than a desired state. It's like with our private and public code. In the early stages, we changed all classes in a backward-compatible way, regardless of whether they are API or not. But now we follow BC only for code marked as API. The same should be for POST request. If we follow BC for POST request then we should document a list of maintained parameters with their descriptions like for WEB API. My opinion is that only WEB API requests should be under BC and possibly some shortlist of other requests. At least it can be our desired state.

Reply:
I have no input on whether or not we should consider POST requests public API, though if we do a subset it needs to be documented somewhere and annotated in code (both for usage/development and semver tooling purposes).

If we do have POST requests as part of the API, I would apply the same rules to them as to API functions in PHP: adding required params major, optional params minor, removing any params major according to standard semver rules.